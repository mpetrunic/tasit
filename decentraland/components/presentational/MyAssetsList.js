import React from "react";
import { FlatList, StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import Colors from "@constants/Colors";
import LargeText from "@presentational/LargeText";

// Note: Changing to PureComponent for performance boost
// It is possible to still use function component with React.memo HoC
// See more:
// https://reactjs.org/docs/react-api.html#reactpurecomponent
// https://medium.com/groww-engineering/stateless-component-vs-pure-component-d2af88a1200b
export default class MyAssetsList extends React.PureComponent {
  render() {
    const { myAssetsList, renderItem } = this.props;
    const { length: listAmount } = myAssetsList;
    const withoutAssets = listAmount === 0;
    return withoutAssets ? (
      <View style={styles.emptyContainer}>
        <LargeText>{`You haven't bought any land yet.`}</LargeText>
      </View>
    ) : (
      <FlatList
        data={myAssetsList}
        style={styles.container}
        renderItem={renderItem}
        keyExtractor={item => item.id}
      />
    );
  }
}

MyAssetsList.propTypes = {
  renderItem: PropTypes.func.isRequired,
  myAssetsList: PropTypes.array.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundColor,
  },
  emptyContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
