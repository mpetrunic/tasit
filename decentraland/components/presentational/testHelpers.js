import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
