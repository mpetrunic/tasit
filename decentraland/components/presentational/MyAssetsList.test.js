import React from "react";
import { shallow } from "enzyme";
import MyAssetsList from "./MyAssetsList";
import { estate } from "./testHelpers";

describe("MyAssetsList", () => {
  describe("renders the component", () => {
    const myAssetRenderer = () => {};
    it("without assets", async () => {
      const myAssetsList = [];
      expect(
        shallow(
          <MyAssetsList
            myAssetsList={myAssetsList}
            renderItem={myAssetRenderer}
          />
        )
      ).toMatchSnapshot();
    });

    it("with assets", async () => {
      const myAssetsList = [estate];
      expect(
        shallow(
          <MyAssetsList
            myAssetsList={myAssetsList}
            renderItem={myAssetRenderer}
          />
        )
      ).toMatchSnapshot();
    });
  });
});
