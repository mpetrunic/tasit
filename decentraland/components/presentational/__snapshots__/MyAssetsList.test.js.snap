// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAssetsList renders the component with assets 1`] = `
<FlatList
  data={
    Array [
      Object {
        "id": "123",
        "img": "https://api.decentraland.org/v1/estates/1/map.png",
        "name": "Sample Estate",
        "type": "ESTATE",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`MyAssetsList renders the component without assets 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LargeText>
    You haven't bought any land yet.
  </LargeText>
</Component>
`;
